server/test/integration/vitalsTest/handleHeartbeatTest.js

```diff
@@ -56,6 +61,54 @@ async function lowBatteryHeartbeat(coreId) {
 }
 
 describe('vitals.js integration tests: handleHeartbeat', () => {
+  describe('for a request with an incorrect API key', () => {
+    beforeEach(async () => {
+      await db.clearTables()
+
+      const client = await factories.clientDBFactory(db)
+      await locationDBFactory(db, {
+        locationid: testLocation1Id,
+        radarCoreId: radar_coreID,
+        clientId: client.id,
+      })
+
+      sandbox.stub(braveAlerter, 'startAlertSession')
+      sandbox.stub(braveAlerter, 'sendSingleAlert')
+      sandbox.spy(helpers, 'logSentry')
+      sandbox.stub(db, 'logSensorsVital')
+    })
+
+    afterEach(async () => {
+      await db.clearTables()
+      sandbox.restore()
+    })
+
+    it('should return 200 for the valid request', async () => {
+      const badRequest = {
+        coreid: radar_coreID,
+        data: `{"doorMissedMsg": 0, "doorLowBatt": true, "doorLastMessage": 1000, "resetReason": "NONE", "states":[]}`,
+        api_key: badpassword,
+      }
+      const response = await chai.request(server).post('/api/heartbeat').send(badRequest)
```

Comment:
I suggest that you factor this part out and put it in a descriptively-named function near the top of the file like what was done with `normalHeartbeat`.

Reply:
sounds good! thank you : )